eng/pipelines/migration.yml

```diff
@@ -0,0 +1,120 @@
+parameters:
```

Comment:
I understand this `migration.yml` is a source of a pipeline that we will use for "string find-and-replace across multiple repositories". If I am correct on that, may I suggest a different name? Something like `string-replace-across-repos.yml` would be more appropriate. My main concern here is that `migration.yml` is a very generic sounding name, that can mean many things. If we ever need to do some other kind of migration, we will have name clash and it may lead to confusion on which migration file is responsible for which kind of migration.

Reply:
Sync with Wes, this pipeline is specific for agent pool migration. If we have generic use in future, we can pull out the common logic. We renamed it to agent-pool-migration.yml. The PR title and branch name are also related to the concept of migration.